kctf-operator/pkg/controller/challenge/challenge_update.go

```diff
@@ -0,0 +1,64 @@
+// File that ensures if all configurations are correctly set
+package challenge
+
+import (
+	kctfv1alpha1 "github.com/google/kctf/pkg/apis/kctf/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+)
+
+func UpdateDeployed(challenge *kctfv1alpha1.Challenge, deployment *appsv1.Deployment) bool {
+	// First, ensure if the challenge is ready and, if not, set replicas to 0
+	// TODO: check if horizontal autoscaling is enabled
+	if challenge.Spec.Deployed == false && *deployment.Spec.Replicas != 0 {
+		var numReplicas int32 = 0
```

Comment:
nit: I saw that others use helper functions that return a pointer, so that you can inline this. If you expect you would need a pointer to int32 often you might consider adding such a helper.

Example: https://github.com/elastic/cloud-on-k8s/blob/master/pkg/utils/pointer/numeric.go#L8

Reply:
(it's not blocking; you can add TODO if you decide to implement it)